src/app/server/Server.h

```diff
@@ -59,11 +62,80 @@ using ServerTransportMgr = chip::TransportMgr<chip::Transport::UDP
 #endif
                                               >;
 
+struct ServerInitParams
+{
+    ServerInitParams()          = default;
+    virtual ~ServerInitParams() = default;
+
+    // Not copyable
+    ServerInitParams(const ServerInitParams &) = delete;
+    ServerInitParams & operator=(const ServerInitParams &) = delete;
+
+    // App delegate to handle some lifecycle events
+    AppDelegate * appDelegate = nullptr;
+    // Port to use for Matter commissioning/operational traffic
+    uint16_t operationalServicePort = CHIP_PORT;
+    // Port to use for UDC if supported
+    uint16_t userDirectedCommissioningPort = CHIP_UDC_PORT;
+    // Interface on which to run daemon
+    Inet::InterfaceId interfaceId = Inet::InterfaceId::Null();
+
+    // Persistent storage delegate: MUST be injected. Used to maintain storage by much common code.
+    PersistentStorageDelegate * persistentStorageDelegate = nullptr;
+    // Group data provider: MUST be injected. Used to maintain critical keys such as the Identity
+    // Protection Key (IPK) for case
+    Credentials::GroupDataProvider * groupDataProvider = nullptr;
+    // Access control delegate: MUST be injected. Used to look-up access control rules
+    Access::AccessControl::Delegate * accessDelegate = nullptr;
+};
+
+// Transitional pre-injection version of ServierInitParams
+struct CommonCaseDeviceServerInitParams : public ServerInitParams
+{
+    CommonCaseDeviceServerInitParams() = default;
+
+    // Not copyable
+    CommonCaseDeviceServerInitParams(const CommonCaseDeviceServerInitParams &) = delete;
+    CommonCaseDeviceServerInitParams & operator=(const CommonCaseDeviceServerInitParams &) = delete;
+
+    virtual CHIP_ERROR InitBeforeServerInit()
```

Comment:
Worth documenting the flow here: that you call this, then you set whatever you want on this thing, then you call Server::Init.

Reply:
Done